Lecture 7


In [27]:
#Report Card Dictionary

report_card_dictionary ={"student_names":["Jay","Aryan","Sahil","Riya","Shubh"],
                        "math_scores": [20,56,76,78,90],
                         "chemistry_scores":[32,56,78,23,100],
                         "information_technology_scores":[34,36,48,100,45],
                         "physics_scores":[6,88,58,90,25] }
report_card_dictionary

{'student_names': ['Jay', 'Aryan', 'Sahil', 'Riya', 'Shubh'],
 'math_scores': [20, 56, 76, 78, 90],
 'chemistry_scores': [32, 56, 78, 23, 100],
 'information_technology_scores': [34, 36, 48, 100, 45],
 'physics_scores': [6, 88, 58, 90, 25]}

In [16]:
#Setup
import pandas as pd

In [29]:
#Converting Dictionary to DataFrame
report_card_df = pd.DataFrame(report_card_dictionary)
report_card_df


student_names  math_scores  chemistry_scores  information_technology_scores  \
0           Jay           20                32                             34   
1         Aryan           56                56                             36   
2         Sahil           76                78                             48   
3          Riya           78                23                            100   
4         Shubh           90               100                             45   

   physics_scores  
0               6  
1              88  
2              58  
3              90  
4              25

In [30]:
#First two rowsin DataFrame
report_card_df.head(2)

student_names  math_scores  chemistry_scores  information_technology_scores  \
0           Jay           20                32                             34   
1         Aryan           56                56                             36   

   physics_scores  
0               6  
1              88

In [31]:
#Sample of two rows in our DataFrame
report_card_df.sample(3)

student_names  math_scores  chemistry_scores  information_technology_scores  \
1         Aryan           56                56                             36   
4         Shubh           90               100                             45   
3          Riya           78                23                            100   

   physics_scores  
1              88  
4              25  
3              90

In [32]:
#Create a description of our numeric columns
report_card_df.describe()

math_scores  chemistry_scores  information_technology_scores  \
count      5.00000          5.000000                        5.00000   
mean      64.00000         57.800000                       52.60000   
std       27.45906         31.893573                       27.14406   
min       20.00000         23.000000                       34.00000   
25%       56.00000         32.000000                       36.00000   
50%       76.00000         56.000000                       45.00000   
75%       78.00000         78.000000                       48.00000   
max       90.00000        100.000000                      100.00000   

       physics_scores  
count        5.000000  
mean        53.400000  
std         37.453972  
min          6.000000  
25%         25.000000  
50%         58.000000  
75%         88.000000  
max         90.000000

In [58]:
#Create a column called total score

report_card_df["total_score"] = report_card_df["math_scores"] + report_card_df["chemistry_scores"] + report_card_df["information_technology_scores"] + report_card_df["physics_scores"]
report_card_df.head(5)

student_names  math_scores  chemistry_scores  information_technology_scores  \
0           Jay           20                32                             34   
1         Aryan           56                56                             36   
2         Sahil           76                78                             48   
3          Riya           78                23                            100   
4         Shubh           90               100                             45   

   physics_scores  total_score  average_score status character grade  \
0               6           92          23.00   fail             NaN   
1              88          236          59.00   pass             NaN   
2              58          260          65.00   pass             NaN   
3              90          291          72.75   pass             NaN   
4              25          260          65.00   pass             NaN   

  character_grade  
0               F  
1               C  
2               C  
3               B  
4               C


Create a column called average score


In [34]:
report_card_df["average_score"] = report_card_df["total_score"]/ 4
report_card_df

student_names  math_scores  chemistry_scores  information_technology_scores  \
0           Jay           20                32                             34   
1         Aryan           56                56                             36   
2         Sahil           76                78                             48   
3          Riya           78                23                            100   
4         Shubh           90               100                             45   

   physics_scores  total_score  average_score  
0               6           92          23.00  
1              88          236          59.00  
2              58          260          65.00  
3              90          291          72.75  
4              25          260          65.00

In [35]:
#Describe the report card dataframe
report_card_df.describe()


math_scores  chemistry_scores  information_technology_scores  \
count      5.00000          5.000000                        5.00000   
mean      64.00000         57.800000                       52.60000   
std       27.45906         31.893573                       27.14406   
min       20.00000         23.000000                       34.00000   
25%       56.00000         32.000000                       36.00000   
50%       76.00000         56.000000                       45.00000   
75%       78.00000         78.000000                       48.00000   
max       90.00000        100.000000                      100.00000   

       physics_scores  total_score  average_score  
count        5.000000     5.000000       5.000000  
mean        53.400000   227.800000      56.950000  
std         37.453972    78.384948      19.596237  
min          6.000000    92.000000      23.000000  
25%         25.000000   236.000000      59.000000  
50%         58.000000   260.000000      65.000000  
75%         88.000000   260.000000      65.000000  
max         90.000000   291.000000      72.750000

# Insights

- **What is the mean chemistry score?** 62.6
- **What is the lowest physics score?** 25
- **What is the highest total score?** 291
- **What is the mean total score?** 248
- **What is the mean average score?** 62

In [36]:
#Creating a Table called Report Card Insights
report_card_insights_df = report_card_df.describe().transpose()
report_card_insights_df

count    mean        std   min    25%    50%  \
math_scores                      5.0   64.00  27.459060  20.0   56.0   76.0   
chemistry_scores                 5.0   57.80  31.893573  23.0   32.0   56.0   
information_technology_scores    5.0   52.60  27.144060  34.0   36.0   45.0   
physics_scores                   5.0   53.40  37.453972   6.0   25.0   58.0   
total_score                      5.0  227.80  78.384948  92.0  236.0  260.0   
average_score                    5.0   56.95  19.596237  23.0   59.0   65.0   

                                 75%     max  
math_scores                     78.0   90.00  
chemistry_scores                78.0  100.00  
information_technology_scores   48.0  100.00  
physics_scores                  88.0   90.00  
total_score                    260.0  291.00  
average_score                   65.0   72.75

In [37]:
#Creating a CSV File of our Report Card Insights

report_card_insights_df.to_csv("Report Card Insight.csv")


# Lecture 8 Classwork

In [41]:
#Create a column called Status that contains a value that states whether a student passed or failed the wuarter semester
report_card_df.loc[report_card_df["total_score"]>=140, "status"] = "pass"
report_card_df.loc[report_card_df["total_score"]<140, "status"] = "fail"

In [42]:
report_card_df

student_names  math_scores  chemistry_scores  information_technology_scores  \
0           Jay           20                32                             34   
1         Aryan           56                56                             36   
2         Sahil           76                78                             48   
3          Riya           78                23                            100   
4         Shubh           90               100                             45   

   physics_scores  total_score  average_score status  
0               6           92          23.00   fail  
1              88          236          59.00   pass  
2              58          260          65.00   pass  
3              90          291          72.75   pass  
4              25          260          65.00   pass

In [59]:
#Create a column called Character Grade that contains the Character Grade for the student
report_card_df.loc[report_card_df["average_score"]>=90,"character_grade"] = "A"
report_card_df.loc[(report_card_df["average_score"]>=70)&(report_card_df["average_score"]<90),"character_grade"] = "B"
report_card_df.loc[(report_card_df["average_score"]>=50)&(report_card_df["average_score"]<70),"character_grade"] = "C"
report_card_df.loc[(report_card_df["average_score"]>=35)&(report_card_df["average_score"]<50),"character_grade"] = "D"
report_card_df.loc[report_card_df["average_score"]<35,"character_grade"] = "F"


In [60]:
report_card_df


student_names  math_scores  chemistry_scores  information_technology_scores  \
0           Jay           20                32                             34   
1         Aryan           56                56                             36   
2         Sahil           76                78                             48   
3          Riya           78                23                            100   
4         Shubh           90               100                             45   

   physics_scores  total_score  average_score status character grade  \
0               6           92          23.00   fail             NaN   
1              88          236          59.00   pass             NaN   
2              58          260          65.00   pass             NaN   
3              90          291          72.75   pass             NaN   
4              25          260          65.00   pass             NaN   

  character_grade  
0               F  
1               C  
2               C  
3               B  
4               C